plugins/modules/cloud/misc/terraform.py

```diff
@@ -518,6 +535,10 @@ def main():
     else:
         outputs = json.loads(outputs_text)
 
+    # delete the generated variables.tf file after command returns
+    if declare_variables:
+        gen_variables_file.close()
```

Comment:
This does not happen in case the code errors out for whatever reason. How about putting a large `try / finally` block around the above code (starting in line 479) with this in the `finally` block?